CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
> In fact this introduces an inconsistency in the Cabal source tree corner case

I barely what this case actually is. Is it when we query `Cabal` package using cabal-helper?

> What is the use-case for this even? So far the API contract is "ask me for stuff and I'll make sure I can tell you stuff" so we don't need to worry about any of this but now you're opening up that can of worms.

We talked about this some time ago. I plan on using cabal-helper from the IDE. Compiling helper takes time, it is far from instantaneous, and user might not be prepared of what's going on behind the scene. It also creates files in user directory, which may also come unnoticed by him. I want IDE to inform user about both these things - that helper is going to be compiled for the first use and where it is going to be stored.